### PR TITLE
SQS: Fix onQueue

### PR DESCRIPTION
**Pull request by erikd-ambiata**
The `onQueue` function was simply calling `createQueue` but if a
queue with the specified name already existed with a different
VisibiltyTimeout an error was being thrown.

Fix this by making `onQueue` call `getQueueByName` which tries to
get the queue and if it doesn't exist, create it.

! @nhibberd @olorin 

Still not sure if this is the best solution.

Update: Maybe we should rename the existing `createQueue` and give that name to `getQueueByName`.
/jury approved @nhibberd @olorin

### CODE REVIEW

File: mismi-sqs/src/Mismi/SQS/Commands.hs [hunk 1]
```diff
@@ -44,11 +44,21 @@ createQueue q v = do
...
     (res ^. cqrsQueueURL)
   where
-    -- If queue alsready exists (and has different VisibilityTimeout)
+    -- If queue already exists (and has different VisibilityTimeout)
     handleExists = handling _QueueNameExists $ \_ ->
       -- Get existing queue (using default parameters)
       send $ A.createQueue (renderQueueName q)
 
+
+getQueueByName :: QueueName -> Maybe Int -> AWS QueueUrl
```
Comment (nhibberd) [inline mismi-sqs/src/Mismi/SQS/Commands.hs]: One downside here is that in the happy path we are doing an extra api call

**Comment by nhibberd** [comment]
⚡️

File: mismi-sqs/src/Mismi/SQS/Commands.hs (hunk 1 above)
Comment (erikd-ambiata) [reply mismi-sqs/src/Mismi/SQS/Commands.hs]: Yeah, its a matter of what is most common, getting an existing queue or creating a new one. I suspect its the former.

**Comment by charleso** [comment]
FWIW I'm still not clear what actually broke our usage of SQS with the amazonka upgrade. That said this seems like the best way to fix in the absence of being able to reproduce it.

**Comment by olorin** [comment]
👍 LGTM.

**Pull request merged**